sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,87 @@ def is_consistent(self):
         # dimensions
         # in vector language: the set of vectors do not form a basis
         return self.inv_can_transf_matrix.is_square
+
+    def buckingham_pi_theorem(self, list_of_derived_quantities):
+        """
+        list_of_derived_quantities is the input of derived quantities in the form of a list of tuples (quantity_name, dims) where quantity_name
+        is the name of the physical quantity and dims is an instance of Dimension class which possesses the dimensions of the quantity. This function
+        is designed to take in a set of physical variables (all essentially dimensionful) and returns one set of all possible exponents,
+        as guided by the Buckingham's pi theorem (raise the list of quantities to the list of exponents to get the dimensionless numbers).
+
+        One of the next functions get_dimensionless_numbers() is used to produce the requisite result in the form of a list of instances of Dimension
+        class signifying the actual dimensionless quantities.
+
+        For example, consider an RLC circuit with resistance ```$R$```, inductance ```$L$``` and capacitance ```$C$``` connected to a constant
+        voltage source ```$V_{amp}$```, where current ```$I(t)$``` is given by the equation (https://en.wikipedia.org/wiki/RLC_circuit).
+
+        ```
+        \\begin{equation*}
+        \\frac{d^2 I}{d t^2} + \\frac{R}{L} \\frac{d I(t)}{d t} + \\frac{1}{LC}I(t) = 0
+        \\end{equation*}
+        ```
+
+        In the above equation, the solution could admit a current of amplitude ```$I_{amp}$``` and a scale of angular frequency ```$\\omega$```.
+
+        >>> from sympy.physics.units import length, mass, time, current
+        >>> from sympy.physics.units.systems.si import dimsys_SI
+        >>> resistance = mass*length**2*time**(-3)*current**(-2)
+        >>> inductance = mass*length**2*time**(-2)*current**(-2)
+        >>> capacitance = mass**(-1)*length**(-2)*time**4*current**2
+        >>> voltage_amp = mass*length**2*time**(-3)*current**(-1)
+        >>> current_amp = current
+        >>> omega = time**(-1)
+        >>> list_of_quantities = [('R', resistance),('L',inductance),('C',capacitance),('V_amp',voltage_amp),('I_amp',current_amp),('omega',omega)]
```

Comment:
Would it be easier to just pass the dimensions? For example, `list_of_quantities = [resistance, inductance, capacitance, ... ]` instead of tuples with their string abbreviations? Does the algorithm really need to know that `'R'` is the abbreviation for `resistance`? I think this info is not necessary.

When creating a new API, it's important to make it as simple as you can, so that new SymPy users will find it faster to learn how to use it. If the API is too complicated, people will probably not use this code, which is a pity!

(By the way, SymPy is trying not to change the API of code that has already been released, so it's important to create a good API at the first Pull Request, I know this makes things longer, but I hope you understand the problem)

Reply:
Hi @Upabjojr Francesco, sorry I had exams on the way, so it took me a while to come up with solutions. Finally I have managed to reduce the parameters to the buckingham_pi_theorem() function to a simple list, and you can check out the new edition in the recent commits. Actually I tried a lot on the make_html but that did not work in my system somehow. And also, I am currently seeing a new test by the name CircleCI, which's not going through. Could you please help me?

Thanks in advance
Shrohan

Reply:
Hi @Upabjojr Francesco, would like to have some of your final reviews on this patch, if at all it could be merged into the next release (or merge of this pull request). Currently I feel there is only some issue with the CircleCI which I could not figure out at all. And there were some other relevant pieces of advice which have also been incorporated so ....